src/OrbitGl/ThreadTrack.cpp

```diff
@@ -446,11 +446,17 @@ void ThreadTrack::OnCaptureComplete() {
 }
 
 [[nodiscard]] static inline uint64_t GetNextPixelBoundaryTimeNs(
-    float world_x, const internal::DrawData& draw_data) {
-  float normalized_x = (world_x - draw_data.track_start_x) / draw_data.track_width;
-  int pixel_x = static_cast<int>(
-      ceil(normalized_x * draw_data.viewport->WorldToScreenWidth(draw_data.track_width)));
-  return draw_data.min_tick + pixel_x * draw_data.ns_per_pixel;
+    uint64_t time_ns, const internal::DrawData& draw_data) {
+  uint64_t ns_from_start = time_ns - draw_data.min_tick;
+  uint64_t total_ns = draw_data.max_tick - draw_data.min_tick;
+  uint64_t num_pixels_on_track = draw_data.viewport->WorldToScreenWidth(draw_data.track_width);
+
+  // Making a ceiling with integers to not lose precision. Giving a maximum number of 4000 pixels
+  // width in a screen, we can capture for more than a month without having overflow.
+  uint64_t next_pixel = ((ns_from_start * num_pixels_on_track) + total_ns - 1) / total_ns;
+
+  uint64_t next_pixel_ns_from_start = total_ns * next_pixel / num_pixels_on_track;
```

Comment:
On the above line, we divide the value by `total_ns` but in the next line, we multiply the value by `total_ns`.
It's a good idea for improving the readability. But as here our may concern is the precision, I prefer to only have `next pixel` in the comments, for example:

```cpp
// We compute next_pixel_ns_from_start by using:
//     uint64_t next_pixel = ((ns_from_start * num_pixels_on_track) + total_ns - 1) / total_ns;
//     uint64_t next_pixel_ns_from_start = total_ns * next_pixel / num_pixels_on_track;
 uint64_t next_pixel_ns_from_start = ((ns_from_start * num_pixels_on_track) + total_ns - 1) / num_pixels_on_track;
```

Reply:
As discussed offline, every step was part of being precise. 

The first division is the way to make a ceiling in integers:
ceiling(A / B) = (A + B - 1) / B;

I've added a comment below to explain why we are multiplying again. I think it should be enough.

Reply:
If that is the case, we can use ceiling(A/B) = 1 + (A-1) / B to avoid the overflowing of (A+B), as long as A != 0

Reply:
Thanks for the suggestion. A should be != 0 and with your proposal we will gain a little more time (between 20 and 30 minutes), but I will leave as it is because I don't think that we could reach that time and I'm used to read this formula for ceiling, so it's easier to read for me at least.